File: app/local_plugins/agent_ast_translate/android/src/main/kotlin/com/aiagent/agent_ast_translate/AstTranslateAgentFlutterPlugin.kt
package com.aiagent.agent_ast_translate

import android.util.Log
import com.aiagent.plugin_interface.NativeAgentRegistry
import io.flutter.embedding.engine.plugins.FlutterPlugin

/**
 * AstTranslateAgentFlutterPlugin — AST Translate Agent Flutter 插件入口
 *
 * 在 onAttachedToEngine 时注册 "ast" Agent 类型到 NativeAgentRegistry。
 * agents_server 通过 NativeAgentRegistry.create("ast") 创建 AstTranslateAgentSession 实例。
 */
class AstTranslateAgentFlutterPlugin : FlutterPlugin {

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        NativeAgentRegistry.register("ast") { AstTranslateAgentSession() }
        Log.d("AstTranslateAgentPlugin", "Registered NativeAgent type=ast")
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}

File: apps/client/local_plugins/agents/agent_ast_translate/ios/Classes/AstTranslateAgentFlutterPlugin.swift
import Flutter
import ai_plugin_interface
import os.log

/// AST translate agent Flutter plugin entry.
///
/// Registers a factory for the `"ast-translate"` agent type with the global
/// `NativeAgentRegistry`. `agents_server` instantiates a session via
/// `NativeAgentRegistry.shared.create("ast-translate")` when the Dart side
/// calls `createAgent(agentType:"ast-translate", …)`.
public class AstTranslateAgentFlutterPlugin: NSObject, FlutterPlugin {
    private static let log = OSLog(subsystem: "com.aiagent.agent_ast_translate", category: "Plugin")

    public static func register(with registrar: FlutterPluginRegistrar) {
        NativeAgentRegistry.shared.register("ast-translate") { AstTranslateAgentSession() }
        os_log("Registered NativeAgent type=ast-translate", log: log, type: .debug)
    }
}
